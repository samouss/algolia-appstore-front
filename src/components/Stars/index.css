.Stars > i {
  margin-right: 1px;

  &:last-child {
    margin-right: 0;
  }
}
