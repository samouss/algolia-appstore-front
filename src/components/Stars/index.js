/* eslint-disable react/no-array-index-key */

import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Star = () => (
  <i className="fa fa-star" aria-hidden="true" />
);

const StarEmpty = () => (
  <i className="fa fa-star-o" aria-hidden="true" />
);

const StarHalf = () => (
  <i className="fa fa-star-half-o" aria-hidden="true" />
);

const Stars = ({ value, on, color }) => (
  <div
    styleName="Stars"
    style={{ color }}
  >
    {new Array(on).fill(null).map((_, index) => {
      const next = index + 1;

      return (index < value && value < next) ? <StarHalf key={index} />
        : (next <= value) ? <Star key={index} />
          : <StarEmpty key={index} />;
    })}
  </div>
);

Stars.propTypes = {
  value: PropTypes.number.isRequired,
  on: PropTypes.number.isRequired,
  color: PropTypes.string,
};

Stars.defaultProps = {
  color: '',
};

export default Stars;
