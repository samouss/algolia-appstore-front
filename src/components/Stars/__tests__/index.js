import React from 'react';
import { shallow } from 'enzyme';
import Stars from '../index';

describe('<Stars />', () => {
  const defaultProps = {
    value: 0,
    on: 5,
  };

  it('expect to render value 0', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render value 3', () => {
    const props = {
      ...defaultProps,
      value: 3,
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render value 3.5', () => {
    const props = {
      ...defaultProps,
      value: 3.5,
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render value 5', () => {
    const props = {
      ...defaultProps,
      value: 5,
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render value 5 on 10', () => {
    const props = {
      ...defaultProps,
      value: 5,
      on: 10,
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with custom color', () => {
    const props = {
      ...defaultProps,
      color: 'red',
    };

    const component = shallow(
      <Stars
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
