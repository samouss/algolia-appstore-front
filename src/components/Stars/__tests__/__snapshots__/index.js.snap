// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stars /> expect to render value 0 1`] = `
<div
  style={
    Object {
      "color": "",
    }
  }
  styleName="Stars"
>
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Stars /> expect to render value 3 1`] = `
<div
  style={
    Object {
      "color": "",
    }
  }
  styleName="Stars"
>
  <Star />
  <Star />
  <Star />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Stars /> expect to render value 3.5 1`] = `
<div
  style={
    Object {
      "color": "",
    }
  }
  styleName="Stars"
>
  <Star />
  <Star />
  <Star />
  <StarHalf />
  <StarEmpty />
</div>
`;

exports[`<Stars /> expect to render value 5 1`] = `
<div
  style={
    Object {
      "color": "",
    }
  }
  styleName="Stars"
>
  <Star />
  <Star />
  <Star />
  <Star />
  <Star />
</div>
`;

exports[`<Stars /> expect to render value 5 on 10 1`] = `
<div
  style={
    Object {
      "color": "",
    }
  }
  styleName="Stars"
>
  <Star />
  <Star />
  <Star />
  <Star />
  <Star />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Stars /> expect to render with custom color 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
  styleName="Stars"
>
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;
